crates/sui-core/src/consensus_handler.rs

```diff
@@ -130,6 +132,52 @@ impl ConsensusHandlerInitializer {
     }
 }
 
+mod additional_consensus_state {
+    use super::*;
+    /// AdditionalConsensusState tracks any in-memory state that is retained by ConsensusHandler
+    /// between consensus commits. Because of crash recovery, using such data is inherently risky.
+    /// In order to do this safely, we must store data from a fixed number of previous commits.
```

Comment:
it doesn't necessarily have to be a fixed number of previous commits--it just has to be deterministic, right?

for example if we decided to organize deferral into N-commit blocks and zeroed it at the start of each, then you might replay 2*N-1 commits and then start recording from zero at the next multiple of N - but the number of commits from which data is stored could be anywhere from [0, N)

Reply:
yes i suppose that's right, but its just a special case of considering a fixed number and ignoring some of them, right?

Reply:
Yes, I brought this up just bc I felt the wording in the comment was confusing in this way (suggesting the stored data had to be from fixed number of commits, and not simply that the number of commits observed had to be fixed)